frontend/src/App.res

```diff
@@ -131,35 +200,75 @@ let make = () => {
   | Data(data)
   | PartialData(data, _) =>
     let benchmarks: array<GetBenchmarks.t_benchmarks> = data.benchmarks
-    let pulls = collectPulls(benchmarks)
-
-    let (main, mainTitle) = {
-      switch String.split_on_char('/', url.hash) {
-      | list{""} => (<BenchmarkResults synchronize benchmarks />, "master")
-      | list{"", "pull", pullNumberStr} =>
-        switch Belt.Int.fromString(pullNumberStr) {
-        | Some(pullNumber) => (
-            <BenchmarkResults synchronize pullNumber benchmarks />,
-            "#" ++ pullNumberStr,
-          )
-        | None => (<h1> {Rx.string("Invalid pull number: " ++ pullNumberStr)} </h1>, "Not found")
+    let repos = collectRepos(data.benchmarks)
+
+    switch String.split_on_char('/', url.hash) {
+    | list{""} =>
+      switch Belt.Array.get(repos, 0) {
+      | Some(firstRepo) => {
+          // If no repository is selected, this redirects to the first one.
+          ReasonReact.Router.replace("#/" ++ firstRepo)
+          React.null
         }
-      | _ => (<h1> {Rx.string("Unknown route: " ++ url.hash)} </h1>, "Not found")
+      | None => <div> {"No data were found..."->Rx.string} </div>
       }
+    | list{"", orgName, repoName, ...rest} => {
+        let selectedRepo = repos->Belt.Array.getBy(repo => repo == orgName ++ "/" ++ repoName)
+        switch selectedRepo {
+        | None => <div> {"This repo does not exist!"->Rx.string} </div>
+        | Some(selectedRepo) =>
+          let pulls = collectPullsForRepo(~repo=selectedRepo, benchmarks)
+          switch rest {
+          | list{"pull", pullNumberStr} =>
+            switch Belt.Int.fromString(pullNumberStr) {
+            | None =>
+              <div> {("Pull request must be an integer. Got: " ++ pullNumberStr)->Rx.string} </div>
+            | Some(selectedPull) =>
+              if pulls->Belt.Array.some(((pullNr, _)) => pullNr == selectedPull) {
+                let benchmarksForPull = collectBenchmarksForPull(
+                  ~repo=selectedRepo,
+                  ~pull=selectedPull,
+                  benchmarks,
+                )
+                <Content
+                  pulls
+                  selectedRepo
+                  repos
+                  benchmarks=benchmarksForPull
+                  selectedPull
+                  startDate
+                  endDate
+                  onSelectDateRange
+                  synchronize
+                  onSynchronizeToggle
+                />
+              } else {
+                <div> {"This pull request does not exist!"->Rx.string} </div>
+              }
+            }
+          | _ =>
```

Comment:
This shouldn't be `_` as that will capture anything except `/pull/xxx`. I think what we want is `list{}`?

Reply:
Indeed, `_` is less restrictive since anything after `org/repo` other than `/pull/xxx` (e.g `org/repo/whatever/else`) would still display data for the repo (instead of showing an "unknown route error").